python/hail/plot/plots.py

```diff
@@ -22,6 +32,19 @@ def histogram(data, legend=None, title=None):
     -------
     :class:`bokeh.plotting.figure.Figure`
     """
+    if isinstance(data, Expression):
+        if data._indices.source is not None:
+            agg_f = data._aggregation_method()
+            if range is not None:
+                start = range[0]
+                end = range[1]
+            else:
+                start = agg_f(aggregators.min(data))
```

Comment:
we should execute both of these at the same time. Otherwise, we read all the data twice!

```
start, end = agg_f((hl.agg.min(data), hl.agg.max(data)))
```